pages/blog/zenwave-asyncapi-code-generator.md

```diff
@@ -0,0 +1,348 @@
+    <version>${zenwave.version}</version>
+    <configuration>
+        <inputSpec>classpath:/model/asyncapi.yml</inputSpec>
+        <addCompileSourceRoot>true</addCompileSourceRoot>
+        <addTestCompileSourceRoot>true</addTestCompileSourceRoot>
+    </configuration>
+    <executions>
+        <!-- DTOs -->
+        <execution>
+            <id>generate-asyncapi-dtos</id>
+            <phase>generate-sources</phase>
+            <goals>
+                <goal>generate</goal>
+            </goals>
+            <configuration>
+                <generatorName>jsonschema2pojo</generatorName>
+                <configOptions>
+                    <modelPackage>io.zenwave360.example.core.domain.events</modelPackage>
+                    <!--        <jsonschema2pojo.includeTypeInfo>true</jsonschema2pojo.includeTypeInfo>-->
+                    <jsonschema2pojo.useLongIntegers>true</jsonschema2pojo.useLongIntegers>
+                </configOptions>
+            </configuration>
+        </execution>
+        <!-- Generate PROVIDER -->
+        <execution>
+            <id>generate-asyncapi</id>
+            <phase>generate-sources</phase>
+            <goals>
+                <goal>generate</goal>
+            </goals>
+            <configuration>
+                <generatorName>spring-cloud-streams3</generatorName>
+                <configOptions>
+                    <role>provider</role>
+                    <style>imperative</style>
+                    <transactionalOutbox>mongodb</transactionalOutbox>
+                    <modelPackage>io.zenwave360.example.core.domain.events</modelPackage>
+                    <producerApiPackage>io.zenwave360.example.core.outbound.events</producerApiPackage>
+                    <consumerApiPackage>io.zenwave360.example.adapters.commands</consumerApiPackage>
+                </configOptions>
+            </configuration>
+        </execution>
+    </executions>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.github.zenwave360.zenwave-sdk.plugins</groupId>
+            <artifactId>asyncapi-spring-cloud-streams3</artifactId>
+            <version>${zenwave.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.github.zenwave360.zenwave-sdk.plugins</groupId>
+            <artifactId>asyncapi-jsonschema2pojo</artifactId>
+            <version>${zenwave.version}</version>
+        </dependency>
+    </dependencies>
+</plugin>
+```
+
+![ZenWave StreamCloudStreams AsyncAPI](/img/posts/zenwave/spring-cloud-streams3-generated.png)
+
+### Provider vs Client
+
+Because broker based API definitions are inherently **symmetrical** it's difficult to establish the roles of client/server. ZenWave generates code based on `provider` and `client` roles, where a `provider` "produces events" and "consumes commands". See [API-First with AsyncAPI](API-First-with-AsyncAPI) page for more details on "publish/subscribe", "producer/consumer" and "provider/client" roles.
```

Comment:
```suggestion
Because broker-based API definitions are inherently **symmetrical**, it's difficult to establish the roles of client/server. ZenWave generates code based on `provider` and `client` roles, where a `provider` "produces events" and "consumes commands". See [API-First with AsyncAPI](API-First-with-AsyncAPI) page for more details on "publish/subscribe", "producer/consumer," and "provider/client" roles.
```